src/ecalc/libraries/libecalc/common/tests/input/yaml_types/test_temporal_model.py

```diff
@@ -0,0 +1,25 @@
+from libecalc.input.yaml_types.temporal_model import TemporalModel
+from pydantic import schema_of
+
+
+class TestSchema:
+    def test_temporal_model_schema(self):
+        """
+        Test to make sure temporal model creates the correct schema. We could improve TemporalModel to generate
+        patternProperties, we would also need to change schema_helpers.replace_temporal_placeholder_property_with_legacy_ref
+        """
+        assert schema_of(TemporalModel[str], title="TemporalModel") == {
+            "title": "TemporalModel",
+            "anyOf": [
+                {
+                    "type": "string",
+                },
+                {
+                    "type": "object",
+                    "patternProperties": {
+                        "^\\d{4}\\-(0?[1-9]|1[012])\\-(0?[1-9]|[12][0-9]|3[01])$": {"type": "string"}
+                    },
+                    "additionalProperties": {"type": "string"},
```

Comment:
`additionalProperties` added by pydantic, it does not exist in the old schemas

Reply:
not sure if I understand this. Is it correct? Which case does it cover?

Reply:
It describes the type of additional properties, but not the keys. So it accepts any key, while we want it to only accept the patternProperties. This is an issue we have with variables and temporal models. We'll have to look into it.